src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/multiples/MultiplesDocumentAccessServiceTest.java

```diff
@@ -0,0 +1,182 @@
+import org.mockito.Mockito;
+import uk.gov.hmcts.et.common.model.ccd.items.DocumentTypeItem;
+import uk.gov.hmcts.et.common.model.ccd.types.DocumentType;
+import uk.gov.hmcts.et.common.model.ccd.types.UploadedDocumentType;
+import uk.gov.hmcts.et.common.model.multiples.MultipleData;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+class MultiplesDocumentAccessServiceTest {
+
+    private MultiplesDocumentAccessService multiplesDocumentAccessService;
+
+    private MultipleData multipleData;
+
+    @BeforeEach
+    void setUp() {
+        multiplesDocumentAccessService = new MultiplesDocumentAccessService();
+        multipleData = new MultipleData();
+        multipleData.setDocumentCollection(new ArrayList<>());
+        multipleData.setClaimantDocumentCollection(new ArrayList<>());
+        multipleData.setLegalrepDocumentCollection(new ArrayList<>());
+
+    }
+
+    @Test
+    void testSetMultipleDocumentCollection() {
+        multipleData = new MultipleData();
+        multipleData.setDocumentCollection(new ArrayList<>());
+
+        UploadedDocumentType uploadedDocumentType = new UploadedDocumentType();
+        uploadedDocumentType.setDocumentFilename("random.pdf");
+
+        DocumentType documentType = new DocumentType();
+        documentType.setUploadedDocument(uploadedDocumentType);
+
+        DocumentTypeItem document1 = new DocumentTypeItem();
+        document1.setValue(documentType);
+        document1.setId("1");
+
+        multipleData.getDocumentCollection().add(document1);
+
+        multiplesDocumentAccessService.setMultipleDocumentCollection(multipleData);
+
+        assertEquals(1, multipleData.getDocumentCollection().size());
+        assertTrue(multipleData.getDocumentCollection().contains(document1));
+    }
+
+    @Test
+    void testSetMultipleDocumentCollectionWhenDocumentCollectionIsEmpty() {
+        // Create a mock MultipleData object with an empty document collection
+        MultipleData multipleData = mock(MultipleData.class);
+        when(multipleData.getDocumentCollection()).thenReturn(new ArrayList<>());
+
+        // Call the method to set multiple document collection
+        multiplesDocumentAccessService.setMultipleDocumentCollection(multipleData);
+
+        // Verify that the documentSelect is not set when document collection is empty
+        verify(multipleData, Mockito.never()).setDocumentSelect(any());
+
+        // Assert that the documentSelect is not set when document collection is empty
+        assertNull(multipleData.getDocumentSelect());
```

Comment:
Were these comments AI gen'd? we don't need to be commenting line by line, makes the code longer and risks out of date documentation if code changes in future